src/remote-config/remote-config-api-client.ts

```diff
@@ -107,16 +107,20 @@ export class RemoteConfigApiClient {
   }
 
   public getTemplate(versionNumber?: number | string): Promise<RemoteConfigTemplate> {
-    let path = 'remoteConfig';
-    if (versionNumber) {
-      path += '?versionNumber=' + versionNumber;
+    let requestData: any;
+    if (typeof versionNumber !== 'undefined') {
+      this.validateVersionNumber(versionNumber);
+      requestData = { versionNumber: `${versionNumber}` };
     }
     return this.getUrl()
       .then((url) => {
         const request: HttpRequestConfig = {
           method: 'GET',
-          url: `${url}/${path}`,
-          headers: FIREBASE_REMOTE_CONFIG_HEADERS
+          url: `${url}/remoteConfig`,
+          headers: FIREBASE_REMOTE_CONFIG_HEADERS,
+          ...requestData
```

Comment:
This is not very readable. Change to `data: requestData` once the above changes are made.

Reply:
I used the spread syntax here to avoid sending an empty object for data when no version number is provided. Will update the code to be more readable :)